google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/apis/CloudApiManager.java

```diff
@@ -119,6 +146,175 @@ static void enableApis(Set<CloudLibrary> libraries, CloudProject cloudProject, P
     }
   }
 
+  /**
+   * Creates a new {@link ServiceAccount}, adds the supplied set of {@link Role roles} to it, and
+   * creates and downloads the service account private key to the user's file system.
+   *
+   * @param roles the set of {@link Role} to add to the new service account
+   * @param name the name of the new service account to be created
+   * @param downloadDir the {@link Path} of the download directory of the service account private
+   *     key json file
+   * @param cloudProject the current {@link CloudProject}
+   * @param project the current {@link Project}
+   */
+  static void createServiceAccountAndDownloadKey(
+      Set<Role> roles, String name, Path downloadDir, CloudProject cloudProject, Project project) {
+    Optional<CredentialedUser> user =
+        Services.getLoginService().getLoggedInUser(cloudProject.googleUsername());
+
+    if (!user.isPresent()) {
+      LOG.error("Cannot enable APIs: logged in user not found.");
+      return;
+    }
+
+    ProgressIndicator progress = ProgressManager.getInstance().getProgressIndicator();
+
+    try {
+      int numSteps = roles.isEmpty() ? 3 : 4;
+      double step = 1;
+
+      showProgress(
+          progress,
+          GctBundle.message("cloud.apis.service.account.create.account.progress.message", name),
+          step / numSteps);
+      step++;
+      ServiceAccount serviceAccount = createServiceAccount(user.get(), name, cloudProject);
+
+      if (!roles.isEmpty()) {
+        showProgress(
+            progress,
+            GctBundle.message("cloud.apis.service.account.add.roles.progress.message"),
+            step / numSteps);
+        step++;
+        addRolesToServiceAccount(user.get(), serviceAccount, roles, cloudProject);
+      }
+
+      showProgress(
+          progress,
+          GctBundle.message("cloud.apis.service.account.create.key.progress.message"),
+          step / numSteps);
+      step++;
+      ServiceAccountKey serviceAccountKey = createServiceAccountKey(user.get(), serviceAccount);
+
+      showProgress(
+          progress,
+          GctBundle.message("cloud.apis.service.account.download.key.progress.message"),
+          step / numSteps);
+      Path keyPath = writeServiceAccountKey(serviceAccountKey, downloadDir, cloudProject);
+
+      notifyServiceAccountCreated(project, name, keyPath);
+    } catch (IOException e) {
+      LOG.warn(
```

Comment:
Could we also show this error to the user directly, using the same notification/bubble approach?

Reply:
I'm now actually debating changing this to a log.error so that an IDE level error surfaces and we get notified. This would be in the case of an API failure so it seems like a good case for us to be alerted. wdyt?

Reply:
Oh you mean this would look like plugin level error when it shows report to google? Looks OK, except won' they be offered an option to disable plugin in this dialog, could be a bit harsh?

Reply:
thats true. and we probably don't want to treat things that may fail as a result of transient network issues as an IDE level error. I'll look into surfacing it to the user as a warning instead.